java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -155,15 +164,11 @@ public boolean equals(Object obj) {
       return false;
     }
     MethodYield other = (MethodYield) obj;
-    if (!Arrays.equals(parametersConstraints, other.parametersConstraints)
-      || exception != other.exception
-      || resultIndex != other.resultIndex
-      || varArgs != other.varArgs) {
-      return false;
-    }
-    if (resultConstraint != null) {
-      return resultConstraint.equals(other.resultConstraint);
-    }
-    return other.resultConstraint == null;
+    return Arrays.equals(parametersConstraints, other.parametersConstraints)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)